From 9d1e094990638e4c00b9101f9d56fb16f2e763c8 Mon Sep 17 00:00:00 2001
From: SimonShin <1847401037@qq.com>
Date: Tue, 12 Dec 2023 14:33:38 +0800
Subject: [PATCH] curl support unizip insetad of zlib spec

---
 curl.spec | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/curl.spec b/curl.spec
index 0ce8985..d132d13 100644
--- a/curl.spec
+++ b/curl.spec
@@ -6,7 +6,7 @@
 
 Name:           curl
 Version:        7.79.1
-Release:        25
+Release:        26
 Summary:        Curl is used in command lines or scripts to transfer data
 License:        MIT
 URL:            https://curl.haxx.se/
@@ -78,6 +78,7 @@ Patch64:	backport-CVE-2023-38546.patch
 Patch65:        backport-CVE-2023-46218.patch
 Patch66:        backport-0001-CVE-2023-46219.patch
 Patch67:        backport-0002-CVE-2023-46219.patch
+Patch68:        support-unizip-instead-of-zlib.patch
 
 BuildRequires:  automake brotli-devel coreutils gcc groff krb5-devel
 BuildRequires:  libidn2-devel libnghttp2-devel libpsl-devel
@@ -120,6 +121,8 @@ Header files for libcurl.
 %prep
 %autosetup -n %{name}-%{version} -p1
 
+autoreconf -fi
+
 printf "1112\n1455\n1184\n1801\n1592\n" >> tests/data/DISABLED
 
 # adapt test 323 for updated OpenSSL
@@ -153,10 +156,10 @@ export common_configure_opts="--cache-file=../config.cache \
         --enable-ldap \
         --enable-ldaps \
         --enable-manual \
-        --with-brotli \
         --with-libidn2 \
         --with-libpsl \
-        --with-libssh
+        --with-libssh \
+        --with-unizip
 )
 
 sed -e 's/^runpath_var=.*/runpath_var=/' \
@@ -246,6 +249,9 @@ rm -rf ${RPM_BUILD_ROOT}%{_libdir}/libcurl.la
 %{_mandir}/man3/*
 
 %changelog
+* Fri Dec 08 2023 tanxin <njutanx@163.com> - 7.79.1-26
+- DESC:support unizip instead of zlib
+
 * Fri Dec 08 2023 zhouyihang <zhouyihang3@h-partners.com> - 7.79.1-25
 - Type:CVE
 - CVE:CVE-2023-46218 CVE-2023-46219
-- 
2.27.0

