From 99c585d59fe21a689c4adee8fd82da59f7e171b3 Mon Sep 17 00:00:00 2001
From: tannnxin <njutanx@163.com>
Date: Sat, 9 Dec 2023 18:28:01 +0800
Subject: [PATCH] gnutls support ifm_nettle instead of nettle spec

---
 gnutls.spec | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/gnutls.spec b/gnutls.spec
index f8795d5..da9e847 100644
--- a/gnutls.spec
+++ b/gnutls.spec
@@ -1,6 +1,6 @@
 Name: gnutls
 Version: 3.7.2
-Release: 10
+Release: 11
 Summary: The GNU Secure Communication Protocol Library
 
 License: LGPLv2.1+ and GPLv3+
@@ -14,6 +14,7 @@ Patch3: backport-CVE-2021-4209.patch
 Patch4: backport-01-CVE-2023-0361.patch
 Patch5: backport-02-CVE-2023-0361.patch
 Patch6: backport-CVE-2023-5981-auth-rsa_psk-side-step-potential-side-channel.patch 
+Patch7: support-ifm_nettle-instead-of-nettle.patch 
 
 %bcond_without dane
 %bcond_with guile
@@ -139,8 +140,9 @@ export GUILD
            --disable-dane \
 %endif
            --disable-rpath \
-           --with-default-priority-string="@SYSTEM"
- 
+           --with-default-priority-string="@SYSTEM" \
+           --with-ifm_nettle 
+
 make %{?_smp_mflags} V=1
  
 %if %{with fips}
@@ -220,6 +222,9 @@ rm -f $RPM_BUILD_ROOT%{_libdir}/pkgconfig/gnutls-dane.pc
 %endif
 
 %changelog
+* Sat Dec 09 2023 tanxin <tanxin@163.com> - 3.7.2-11                                                                                                         
+- support ifm_nettle instead of nettle
+
 * Mon Nov 20 2023 xuraoqing <xuraoqing@huawei.com> - 3.7.2-10                                                                                                         
 - fix CVE-2023-5981
 
-- 
2.33.0

