From 9d1f6e62231839c44e525076785478d94378311b Mon Sep 17 00:00:00 2001
From: hugin1_1 <hugin1_1@sina.com>
Date: Sun, 10 Dec 2023 20:20:03 +0800
Subject: [PATCH] support-ifm_gcrypt-in-gnupg-spec

---
 gnupg2.spec | 16 +++++++++++++---
 1 file changed, 13 insertions(+), 3 deletions(-)

diff --git a/gnupg2.spec b/gnupg2.spec
index e2fbba4..48e8ef0 100644
--- a/gnupg2.spec
+++ b/gnupg2.spec
@@ -1,6 +1,6 @@
 Name: gnupg2
 Version: 2.2.32
-Release: 5
+Release: 6
 Summary: Utility for secure communication and data storage
 
 License: GPLv3+
@@ -22,6 +22,8 @@ Patch10: gnupg-2.2.21-coverity.patch
 Patch11: common-Avoid-undefined-behavior-of-left-shift-operat.patch
 Patch12: backport-CVE-2022-34903.patch
 Patch13: backport-common-Protect-against-a-theoretical-integer-overflow.patch
+Patch14: support-ifm-gcrypt-instead-of-gcrypt-in-gnupg2.patch
+
 
 BuildRequires: gcc
 BuildRequires: zlib-devel, npth-devel, texinfo
@@ -66,12 +68,16 @@ signing to the base GnuPG package.
 
 sed -i -e 's/"libpcsclite\.so"/"%{pcsclib}"/' scd/scdaemon.c
 
+autoreconf -fi
+
 %build
 %configure \
   --disable-gpgtar \
   --disable-rpath \
   --enable-g13 \
-  --enable-large-secmem
+  --with-ifm-gcrypt \
+  --enable-large-secmem \ 
+
 
 %make_build
 
@@ -92,7 +98,7 @@ ln -sf gnupg.7 %{buildroot}%{_mandir}/man7/gnupg2.7
 rm -f %{buildroot}%{_infodir}/dir
 
 %check
-make check
+make check
 
 %files -f %{name}.lang
 %defattr(-,root,root)
@@ -119,6 +125,10 @@ make check
 
 
 %changelog
+* Fri Dec 8 2023 geziwen <hugin1_1@sina.com> - 2.2.32-6
+- support ifm-gcrypt instead of gcrypt
+
+
 * Wed Jun 14 2023 yixiangzhike <yixiangzhike007@163.com> - 2.2.32-5
 - backport upstream patch
 
-- 
2.33.0

