From 8cfd43420c13f3c71662838b4aa783d66a7b936f Mon Sep 17 00:00:00 2001
From: huangduirong <huangduirong@huawei.com>
Date: Fri, 3 Nov 2023 08:42:30 +0800
Subject: [PATCH] libvhtp spec

---
 libevhtp.spec | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/libevhtp.spec b/libevhtp.spec
index 970e798..42c8c21 100644
--- a/libevhtp.spec
+++ b/libevhtp.spec
@@ -2,7 +2,7 @@
 
 Name:     	libevhtp
 Version:  	1.2.18
-Release:  	6
+Release:  	7
 Summary:  	Libevent based HTTP API.
 
 License:  	BSD-3-Clause
@@ -14,8 +14,9 @@ Patch9002:  0003-close-open-ssl.-we-do-NOT-use-it-in-lcrd.patch
 Patch9003:  0004-Use-shared-library-instead-static-one.patch
 Patch9004:  0005-libevhtp-add-securce-compile-options.patch
 Patch9005:  0006-libevhtp-add-gcov-compile-options.patch
+Patch9006:  ifm-use-libhv-instead-of-libev.patch
 
-BuildRequires: 	git gcc-c++ cmake libevent-devel
+BuildRequires: 	git gcc-c++ cmake 
 
 %description
 Libevent based HTTP API.Libevent's http interface was created as a JIT server, never meant
@@ -38,7 +39,7 @@ applications that want to make use of %{name}.
 %build
 mkdir -p build
 cd build
-%cmake -DEVHTP_BUILD_SHARED=ON -DEVHTP_DISABLE_SSL=ON -DLIB_INSTALL_DIR=lib ..
+%cmake -DEVHTP_BUILD_SHARED=ON -DEVHTP_DISABLE_SSL=ON -DEVHTP_USE_HV=ON -DLIB_INSTALL_DIR=lib ..
 %make_build
 
 %install
-- 
2.33.0

